enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/TopicController.java

```diff
@@ -0,0 +1,201 @@
+import eu.europeana.enrichment.web.model.EnrichmentTopicRequest;
+import eu.europeana.enrichment.web.service.EnrichmentTopicService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import eu.europeana.enrichment.common.commons.HelperFunctions;
+@RestController
+@EnableCaching
+//@SwaggerSelect
+@Api(tags = "Topic service", description=" ")
+public class TopicController extends BaseRest{
+	
+	@Autowired
+	EnrichmentTopicService enrichmentTopicService;
+	
+	Logger logger = LogManager.getLogger(getClass());
+	
+	
+	
+	/**
+	 * This method represents the /enrichment/topic/ end point,
+	 * where a topic creation request will be processed.
+	 * All requests on this end point are processed here.
+	 * 
+	 * @param wskey
+	 * @return
+	 * @throws Exception 
+	 */
+	@ApiOperation(value = "Create Topic", nickname = "postTopicCreation", notes = "This method stores the topics into the database\n"
+			+ "Mandatory fields: identifier, description and topicTerm")
+	@RequestMapping(value = "/enrichment/topic/", method = {RequestMethod.POST}, produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<String> postTopicCreation (@RequestParam(value = "wskey", required = true) String wskey,
+			@RequestBody(required = false) String text) throws JsonMappingException, JsonProcessingException, HttpException, UnsupportedEntityTypeException
+	{
+		ResponseEntity<String> response = null;
+		try {
+			validateApiKey(wskey);
+		} catch (ApplicationAuthenticationException e) {
+			e.printStackTrace();
+			response = new ResponseEntity<String>("",HttpStatus.UNAUTHORIZED);
+		}
+		
+		// parse JSON 
+		ObjectMapper objectMapper = new ObjectMapper();
+		System.out.println(text);
+		EnrichmentTopicRequest request = objectMapper.readValue(text, EnrichmentTopicRequest.class);
+		
+		
+		
+		// check mandatory fields
+		if (!HelperFunctions.validString(request.topicIdentifier) || HelperFunctions.testNullOrEmpty(request.descriptions) || HelperFunctions.testNullOrEmpty(request.topicTerms) )
+		{
+			response = new ResponseEntity<String>("", HttpStatus.BAD_REQUEST);
+			return response;
+		}
+			
+		// use create topic service
+		TopicEntity te = enrichmentTopicService.createTopic(request);
+		if (te != null)
+		{
+			//TODO: add apiVersion to the generateETag method
+			String etag = generateETag(te.getModifiedDate(), WebFields.JSON_LD_REST);
+			MultiValueMap<String, String> headers = new LinkedMultiValueMap<String, String>(5);
+		    headers.add(HttpHeaders.CONTENT_TYPE, HttpHeaders.CONTENT_TYPE_JSON_UTF8);
+		    headers.add(HttpHeaders.ETAG, etag);
+		    headers.add(HttpHeaders.ALLOW, HttpHeaders.ALLOW_POST);
+		    response = new ResponseEntity<String>(te.toJSON(), headers, HttpStatus.OK);
```

Comment:
Please use the new application wide json serializer from here: eu.europeana.enrichment.common.serializer.JsonLdSerializer, you can see an example in other controllers.